pkg/client/unversioned/request.go

```diff
@@ -164,6 +169,12 @@ func (r *Request) SubResource(subresources ...string) *Request {
 		r.err = fmt.Errorf("subresource already set to %q, cannot change to %q", r.resource, subresource)
 		return r
 	}
+	for _, s := range subresources {
```

Comment:
Fixing this in the client is at best an optimization for non-adversarial clients, and I worry it will lead to us leaving a vulnerability in the server.

If that doesn't persuade you, then I would at least like to not import the entire api/validation package, and I think it might be simpler to check the path once in the function that builds the final URL.